src/checks/table_data_formatting_heirarchical.py

```diff
@@ -0,0 +1,63 @@
+import re
+
+from geopandas import GeoDataFrame
+
+from src.config import ADMIN_BOUNDARY_REGEX, CheckReturnList
+
+
+def main(iso3: str, gdfs: list[GeoDataFrame]) -> CheckReturnList:
+    """Check completeness of an admin boundary by checking column hierarchy.
+
+    Args:
+        iso3: ISO3 code of the current location being checked.
+        gdfs: List of GeoDataFrames, with the item at index 0 corresponding to admin
+        level 0, index 1 to admin level 1, etc.
+
+    Returns:
+        List of check rows to be outputed as a CSV.
+    """
+    check_results = []
+    for admin_level, gdf in enumerate(gdfs):
+        row = {
+            "iso3": iso3,
+            "level": admin_level,
+            "proportion_of_expected_heirarchical_columns": proportion_of_expected_heirarchical_columns(
```

Comment:
Do we want to leave the creation of a 0-1 score for the `scores` module? This check could be simplified if it only counts the total number of hierarchical columns:

```py
"total_heirarchical_columns": len(
    [
        any(column.startswith(f"ADM{level}") for column in gdf.columns)
        for level in range(admin_level + 1)
    ]
),
```

Reply:
You're right, we should prefer to leave the creation of scores to the `scores` module. I was lazy here, happy to refactor. That suggestion also simplifies the code too, thanks